src/main/scala/ch/epfl/bluebrain/nexus/kg/marshallers/ResourceJsonLdCirceSupport.scala

```diff
@@ -0,0 +1,46 @@
+package ch.epfl.bluebrain.nexus.kg.marshallers
+
+import akka.http.scaladsl.marshalling.{Marshaller, ToEntityMarshaller, ToResponseMarshaller}
+import akka.http.scaladsl.marshalling.GenericMarshallers.eitherMarshaller
+import akka.http.scaladsl.marshalling.PredefinedToResponseMarshallers._
+import akka.http.scaladsl.model.{HttpEntity, HttpResponse, StatusCodes}
+import cats.syntax.show._
+import ch.epfl.bluebrain.nexus.commons.http.{JsonLdCirceSupport, RdfMediaTypes}
+import ch.epfl.bluebrain.nexus.kg.config.Vocabulary
+import ch.epfl.bluebrain.nexus.kg.resources.Rejection
+import ch.epfl.bluebrain.nexus.kg.resources.Rejection._
+import io.circe.{Encoder, Json, Printer}
+
+trait ResourceJsonLdCirceSupport extends JsonLdCirceSupport {
+
+  val rejectionEncoder: Encoder[Rejection] = Encoder { rejection =>
+    Json.obj(
+      "@context" -> Json.fromString(Vocabulary.nxv.errorContext.show),
+      "message"  -> Json.fromString(rejection.msg)
+    )
+  }
+
+  final implicit def rejectionToResponseMarshaller(
+      implicit printer: Printer = Printer.noSpaces.copy(dropNullValues = true)): ToResponseMarshaller[Rejection] =
+    Marshaller.withFixedContentType(RdfMediaTypes.`application/ld+json`) { rejection =>
+      val entity = HttpEntity(RdfMediaTypes.`application/ld+json`, printer.pretty(rejectionEncoder(rejection)))
+      rejection match {
```

Comment:
maybe we should pull this in a separate function (it should be easier to see the mapping between a rejection and the status code); it could also be defined as an inner function.

Reply:
The way we did it in [ExceptionHandling](https://github.com/BlueBrain/nexus-admin/blob/master/modules/service/src/main/scala/ch/epfl/bluebrain/nexus/admin/service/handlers/ExceptionHandling.scala#L61) seems clear to me

Reply:
@umbreak I think that's clearer too, but in order to be able to use `ExceptionHandler` from `akka-http` we would have to throw the rejections somewhere and I'm not sure where's the best place to do that.